bootstrap/kubeadm/internal/controllers/kubeadmconfig_controller.go

```diff
@@ -890,7 +941,8 @@ func (r *KubeadmConfigReconciler) storeBootstrapData(ctx context.Context, scope
 			},
 		},
 		Data: map[string][]byte{
-			"value": data,
+			"value":  data,
+			"format": []byte(scope.Config.Spec.Format),
```

Comment:
Could it be that we store an empty string here in the cloud-init case? (as format is not set in that case)

I wonder if we should either:
* handle the case that Format is empty and write `cloud-config`
* only add the "format" field if the format is not empty (I think this should maximize compatibility with the previous behavior)

Reply:
Should we default `spec.format` on CABPK to cloud-init instead?

Reply:
On the webhook? +1

Reply:
Sounds good to me if it's not too intrusive.

Reply:
@vincepri I wonder if we should change the webhook(s) in a separate PR. Imho we should add the defaulting to both KubeadmConfig and KubeadmConfigTemplate and I'm not sure if there will be any side effects. 

I don't think so as we currently don't enforce that KubeadmConfigTemplate is immutable (https://github.com/kubernetes-sigs/cluster-api/issues/4910) and changes in a KubeadmConfigTemplate don't trigger rollouts, but who knows.

Given the size of the current PR. I would prefer for this PR to just set `cloud-config` if `scope.Config.Spec.Format` is empty and then we can add the defaulting a follow-up PR>

Reply:
>Could it be that we store an empty string here in the cloud-init case?

@sbueringer yes. We store whatever is specified in `scope.Config.Spec.Format` including the zero value `""` of that field.

Should I change this behavior or wait for @vincepri to respond?

Reply:
Let's see what Vince says

Reply:
Let's do it separately if we're changing the default from an empty string to cloud-config

Reply:
So I guess we shouldn't even create a format key with empty value for the time being? So there is absolutely no changes to current users.

Reply:
Sounds like a good idea!

I would prefer not setting an empty `format` value

Reply:
As discussed on weekly call, we leave it as it is for now and I created an issue to fix that in the next PR: https://github.com/kubernetes-sigs/cluster-api/pull/4172/files#r765097973